runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientConnectionPool.java

```diff
@@ -446,7 +453,13 @@ private void doReset(
 
     public BindingHandler streamFactory()
     {
-        return this::newStream;
+        BindingHandler newStream = this::newStream;
+        if (!clientConnectionPool)
+        {
+            newStream = streamFactory;
+        }
```

Comment:
This check should be in `KafkaClientFactory` so we pass the correct `streamFactory` and `signaler` to `KafkaClientGroupFactory` instead of this approach.